snapcraft/internal/repo/snaps.py

```diff
@@ -0,0 +1,214 @@
+    snapd and the store data can change in between validation and execution.
+    """
+
+    @classmethod
+    def is_valid_snap(cls, snap):
+        return cls(snap).is_valid()
+
+    @classmethod
+    def is_snap_installed(cls, snap):
+        return cls(snap).installed
+
+    def __init__(self, snap):
+        """Lifecycle handler for a snap of the format <snap-name>/<channel>."""
+        self.name, self.channel = _get_parsed_snap(snap)
+        self._original_channel = self.channel
+        if not self.channel or self.channel == 'stable':
+            self.channel = 'latest/stable'
+
+        # This store information from a local request
+        self._local_snap_info = None
+        # And this stores information from a remote request
+        self._store_snap_info = None
+
+        self._is_installed = None
+        self._is_on_store = None
+
+    @property
+    def installed(self):
+        if self._is_installed is None:
+            self._is_installed = self.get_local_snap_info() is not None
+        return self._is_installed
+
+    @property
+    def on_store(self):
+        if self._is_on_store is None:
+            self._is_on_store = self.get_store_snap_info() is not None
+        return self._is_on_store
+
+    def get_local_snap_info(self):
+        """Returns a local payload for the snap.
+
+        Validity of the results are determined by checking self.installed."""
+        if self._is_installed is None:
+            with contextlib.suppress(exceptions.HTTPError):
+                self._local_snap_info = _get_local_snap_info(self.name)
+        return self._local_snap_info
+
+    def get_store_snap_info(self):
+        """Returns a store payload for the snap.
+
+        Validity of the results are determined by checking self.remote."""
+        if self._is_on_store is None:
+            with contextlib.suppress(exceptions.HTTPError):
+                self._store_snap_info = _get_store_snap_info(self.name)
+        return self._store_snap_info
+
+    def _get_store_channels(self):
+        snap_store_info = self.get_store_snap_info()
+        if not self.on_store:
+            return dict()
+
+        return snap_store_info['channels']
+
+    def get_current_channel(self):
+        current_channel = ''
+        if self.installed:
+            local_snap_info = self.get_local_snap_info()
+            current_channel = local_snap_info['channel']
+            if current_channel == 'stable':
+                current_channel = 'latest/stable'
+        return current_channel
+
+    def is_classic(self):
+        store_channels = self._get_store_channels()
+        return store_channels[self.channel]['confinement'] == 'classic'
+
+    def is_valid(self):
+        """Check if the snap is valid."""
+        if not self.on_store:
+            return False
+        store_channels = self._get_store_channels()
+        return self.channel in store_channels.keys()
+
+    def install(self):
```

Comment:
What if the snap I want to use to build requires devmode? Are we choosing to not support that?

Reply:
not at the moment, check the forum post on the subject

Reply:
I did. It says nothing about devmode, thus my question.

Reply:
right, so it would need a discussion first, right now it shouldn't work. In all fairness, if we discuss this, I'd say it shouldn't just like we are not auto connecting interfaces.